internal/pkg/template/input/step.go

```diff
@@ -1,14 +1,88 @@
 package input
 
-type StepsGroups []StepsGroup
+import (
+	"fmt"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/filesystem"
+)
+
+type StepsGroups []*StepsGroup
+
+func Load(fs filesystem.Fs) (*StepsGroups, error) {
+	f, err := loadFile(fs)
+	if err != nil {
+		return nil, err
+	}
+	return &f.StepsGroups, nil
+}
+
+type StepIndex struct {
+	Step  int
+	Group int
+}
+
+func (g StepsGroups) Indices() map[string]StepIndex {
+	res := make(map[string]StepIndex)
+	for gIdx, group := range g {
+		for sIdx, step := range group.Steps {
+			res[step.Id] = StepIndex{
+				Step:  sIdx,
+				Group: gIdx,
+			}
+		}
+	}
+	return res
+}
+
+func (g StepsGroups) AddInput(input Input, index StepIndex) error {
+	if len(g) < index.Group {
+		return fmt.Errorf("group at index %d not found", index.Group)
+	}
+	if len(g[index.Group].Steps) < index.Step {
+		return fmt.Errorf("step at index %d for group at index %d not found", index.Step, index.Group)
+	}
```

Comment:
To by asi nemělo vůbec nastat díky validaci, ale přišlo mi to trochu divný tady vůbec nekontrolovat. 😃